### PR TITLE
Option to Offload non-technical PR discussion to a Discord Thread

### PR DESCRIPTION
**Pull request by Cyberboss**
This was discussed in #maintainerbus on the Discord. @tgstation/commit-access 

Result of this PR: 

- Add the `Discord Discussion` label to the PR. 
- The bot will create the thread, link it in the PR, and lock the PR. Slowmode set to 1 minute. Auto-archive duration set to max of 1 week.
- The thread will be archived if the PR is merged or closed, unarchived if it's reopened.
- You can also set a join link for the Discord to appear in the GitHub comment. I'll be setting it to our official link at https://tgstation13.org/phpBB/viewforum.php?f=60

image

image
Let the bikeshed commence

### CODE REVIEW

**Comment by Sadboysuss** [comment]
What if I'm a coder and spot something with pr? Then I won't be able to review it or point out the issue? Also, is this restricted to just commit-access or to members too (issue managers are members, for example)?

**Comment by Cyberboss** [comment]
I've setup the required secrets for this except for `vars.DISCORD_DISCUSSIONS_CHANNEL_ID` need to wait for MSO on that one.

**Comment by TheVekter** [comment]
To what degree is the PR locked? Can contributors who aren't directly involved still comment in a technical sense, or is it limited strictly only to the person who opened the PR and maintainers?

**Comment by Cyberboss** [comment]
It's a standard GitHub lock

**Comment by BlueMemesauce** [comment]
Is there a point to this? I guess it will make it easier for people without github accounts to comment on changes, but for everyone that's already on github it just seems like it would be annoying

**Comment by SmArtKar** [comment]
> It's a standard GitHub lock

That restricts the discussion to accounts with write access, meaning that contributors or issue managers won't be able to comment on technical side of the PR or leave reviews regarding code. While this does seem like a good idea, locking the thread outright will definitely lead to some issues.

**Comment by Redrover1760** [comment]
Just... Don't lock the thread. Off-topic comments that aren't technical focused that are applied to the thread, maybe ban those that do it repeatedly.

**Comment by ExcessiveUseOfCobblestone** [comment]
You cannot parse a technical v. non-technical review from the codeside so this is either a good idea with the understanding it will be annoying for people actually reviewing code, or a bad one with the reason it will be annoying for people actually reviewing code.

Im also curious if, Im assuming this is a post-wallening push, it would not have the opposite effect of making people think youre just ignoring stuff by the nature of it not being on the PR itself.

Is the proposed solution to just... leave it open and have a single message saying "hey tech reviews only :)" with human enforcement? There's no perfect solution for getting around not being a fan of the current github UI that doesnt include asking github to make accomodations so in any flavor of this there are going to have to be compromises.

**Comment by TheVekter** [comment]
> Is the proposed solution to just... leave it open and have a single message saying "hey tech reviews only :)" with human enforcement? There's no perfect solution for getting around not being a fan of the current github UI that doesnt include asking github to make accomodations so in any flavor of this there are going to have to be compromises.

This honestly seems like the play. I don't see an issue with just having people report posts that aren't technical in nature and slapping repeat offenders.

**Comment by optimumtact** [comment]
> Is there a point to this? I guess it will make it easier for people without github accounts to comment on changes, but for everyone that's already on github it just seems like it would be annoying

it's limits conversation to people with playtime on tgstation.

File: .github/workflows/discord_discussions.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Discord Discussions
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - reopened
+      - edited
+      - labeled
+      - closed
```
Comment (Mothblocks) [inline .github/workflows/discord_discussions.yml:10]: Why are we doing it for things like edited etc?

Comment (Cyberboss) [reply .github/workflows/discord_discussions.yml:10]: Reopened/closed for thread state change
Opened/Labeled for initialization
Edited for title changes

Comment (Cyberboss) [reply .github/workflows/discord_discussions.yml:10]: The job only runs if the label is present

**Comment by Ghommie** [comment]
...So? Are we waiting for Mothblocks again?

**Comment by Cyberboss** [comment]
Pretty much

**Comment by Mothblocks** [comment]
theres 3 head coders

**Pull request merged**